src/Disciplina/Crypto/Hash/Cryptonite.hs

```diff
@@ -1,18 +1,46 @@
+{-# LANGUAGE CPP              #-}
+{-# LANGUAGE TypeApplications #-}
 
 -- | 'HashFunction' instances for 'Cryptonite' hashes
 
 module Disciplina.Crypto.Hash.Cryptonite
-       ( CryptoniteFunc (..)
+       ( CryptoniteFunc
        ) where
 
 import Universum
 
+import Crypto.Hash (Digest, HashAlgorithm)
 import qualified Crypto.Hash as Crypto
+import Data.ByteArray (ByteArrayAccess, Bytes)
 
-import Disciplina.Crypto.Hash.Class (AbstractHash (..), HashFunction (..))
+import Disciplina.Crypto.Hash.Class (AbstractHash (..), HasAbstractHash (..), HashFunc (..))
+import Disciplina.Crypto.Signing.Class (AbstractPK, AbstractSK, AbstractSig)
 
-data CryptoniteFunc algo = CryptoniteFunc algo
+-- | Tag for choosing a particular hash algorithm from 'Crypto.Hash'
+-- for hashing.
+data CryptoniteFunc algo
```

Comment:
Because we don't know yet what we'll be using. Also, even in Cardano we ended up using two different types of hashes (`addressHash` is `blake224`). Same for signature schemes.